src/Drush/Commands/MultisiteDrushCommands.php

```diff
@@ -0,0 +1,147 @@
+
+  /**
+   * Construct an object of Multisite commands.
+   */
+  public function __construct(private BootstrapManager $bootstrapManager) {
+    parent::__construct();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function createEarly(DrushContainer $drush_container): self {
+    return new static(
+      $drush_container->get('bootstrap.manager')
+    );
+  }
+
+  /**
+   * Execute code before pre-validate site:install.
+   */
+  #[CLI\Hook(type: HookManager::PRE_ARGUMENT_VALIDATOR, target: 'site-install')]
+  public function preValidateSiteInstall(CommandData $commandData): void {
+    if ($this->validateGenerateSettings()) {
+      $uri = $commandData->input()->getOption('uri') ?? 'default';
+      $sitesSubdir = $this->getSitesSubdirFromUri(DRUPAL_ROOT, $uri);
+      $commandData->input()->setOption('sites-subdir', $sitesSubdir);
+      $options = $commandData->options();
+      $this->bootstrapManager->setUri('http://' . $sitesSubdir);
+
+      // Try to get any already configured database information.
+      $this->bootstrapManager->bootstrapMax(DrupalBootLevels::CONFIGURATION, $commandData->annotationData());
+
+      // By default, bootstrap manager boot site from default/setting.php
+      // hence remove the database connection if site is other than default.
+      if (($sitesSubdir && "sites/$sitesSubdir" !== $this->bootstrapManager->bootstrap()->confpath())) {
+        Database::removeConnection('default');
+        $db = [
+          'database' => 'drupal',
+          'username' => 'drupal',
+          'password' => 'drupal',
+          'host' => 'localhost',
+          'port' => '3306',
+        ];
+        $dbSpec = [
+          "drupal" => ["db" => $db]
+        ];
+        if (!($options['db-url'])) {
+          if (EnvironmentDetector::isLocalEnv()) {
+            $db = $this->askDbCredentials($sitesSubdir, $db);
+            $dbSpec["drupal"]["db"] = $db;
+          }
+          $commandData->input()->setOption("db-url",
+            "mysql://" . $db['username'] . ":" . $db['password'] . "@" . $db['host'] . ":" . $db['port'] . "/" . $db['database']
+          );
+        }
+        $settings = new Settings(DRUPAL_ROOT, $sitesSubdir);
+        try {
+          $settings->generate($dbSpec);
+          $this->postGenerateSettings();
+        }
+        catch (SettingsException $e) {
+          $this->io()->warning($e->getMessage());
+        }
+      }
+    }
+  }
+
+  /**
+   * Function to check if multisite should be setup or not.
+   */
+  protected function validateGenerateSettings(): bool {
+    $handlers = $this->getCustomEventHandlers(self::VALIDATE_GENERATE_SETTINGS);
+    $status = TRUE;
+    foreach ($handlers as $handler) {
+      $status = $handler();
+      if (!$status) {
+        return FALSE;
+      }
+    }
+    return $status;
+  }
+
+  /**
+   * Function to run if post generation of settings.php.
+   */
+  protected function postGenerateSettings(string $eventName): void {
```

Comment:
Please remove $eventName, as not being used anywhere.
```suggestion
  protected function postGenerateSettings(): void {
```

Reply:
Done.